src/caffeine/coarray_queries_s.f90

```diff
@@ -22,6 +23,32 @@
   end procedure
 
   module procedure prif_image_index
+    integer :: dim, i
+    integer(c_int) :: prior_size, num_img
+    logical :: invalid_cosubscripts
+
+    invalid_cosubscripts = .false.
+
+    check_subscripts: do i = 1, size(sub)
+       if (sub(i) .lt. coarray_handle%info%lcobounds(i) .or. sub(i) .gt. coarray_handle%info%ucobounds(i)) then
+          invalid_cosubscripts = .true.
+          exit check_subscripts
+       end if
+    end do check_subscripts
+
+    if (.not. invalid_cosubscripts) then
+      image_index = 1 + sub(1) - coarray_handle%info%lcobounds(1)
+      prior_size = 1
+      do dim = 2, size(sub)
+        prior_size = prior_size * (coarray_handle%info%ucobounds(dim-1) - coarray_handle%info%lcobounds(dim-1) + 1)
+        image_index = image_index + (sub(dim) - coarray_handle%info%lcobounds(dim)) * prior_size
+       end do
```

Comment:
```suggestion
       end do
       ```suggestion
      ! Future work: values of prior_size are invariant across calls w/ the same coarray_handle
      !  We could store them in the coarray metadata at allocation rather than redundantly 
      ! computing them here, which would accelerate calls with corank > 1 by removing 
      ! corank multiply/add operations and the loop-carried dependence
      prior_size = 1
```
```

Reply:
Agree. I've added the comment.